### PR TITLE
Fix leaking schemachanged files on master swing

### PR DESCRIPTION
**Pull request by dorinhogea**
The replicants keep track of deleted files so a master swing finds the list ready.
Restarting a node will also populate the list of removed files using logs.

### CODE REVIEW

**Comment by mponomar** [comment]
'send bdb dblist' is also broken.  I broke it, not part of this PR.

**Comment by dorinhogea** [comment]
What steps you do to get :
DB_LOGC->get: invalid LSN: 0/0
[ERROR] log_get_record: failed reading repo log record rc=22

**Comment by dorinhogea** [comment]
The log error happens only for freshly created dbs, that still have log 1.  I have patched that.

File: schemachange/sc_callbacks.c [hunk 1]
```diff
@@ -511,6 +511,13 @@ static int delete_table_rep(char *table)
         return -1;
     }
 
+    /* update the delayed deleted files */
+    rc = bdb_list_unused_files(db->handle, &bdberr, (char *)__func__);
```
Comment (ldennis) [inline schemachange/sc_callbacks.c]: Use bdb_list_unused_files_tran and pass in tran to avoid deadlock

Comment (dorinhogea) [inline schemachange/sc_callbacks.c]: which tran, more precisely?

**Comment by ldennis** [comment]
The latest commit looks good to me

**Comment by dorinhogea** [comment]
thank you

**Comment by ldennis** [comment]
This commit failed the coding style check. You may want to format your changes.

File: bdb/file.c [hunk 2]
```diff
@@ -7622,26 +7630,27 @@ int bdb_have_unused_files(void) { return oldfile_list_empty() != 1; }
 int bdb_purge_unused_files(bdb_state_type *bdb_state, tran_type *tran,
                            int *bdberr)
 {
-    char *munged_name = NULL;
-    int rc;
-    unsigned lognum = 0, lowfilenum = 0;
+  char *munged_name = NULL;
```
Comment (riverszhang89) [inline bdb/file.c]: We indent by 4 spaces, not 2.

Comment (dorinhogea) [inline bdb/file.c]: I did indent by 4 spaces.  The instructions to fix the failure reindented to code to 2 space.

Comment (dorinhogea) [inline bdb/file.c]: (I have followed to instructions to fix the "coding style" error, and it re-indented to 2 spaces.  Waiting for github upload of roborivs code)

Comment (ldennis) [inline bdb/file.c]: which instruction? I think it has `IndentWidth: 4` in the clang-format style session.

Comment (ldennis) [inline bdb/file.c]: I believe roborivers uses something similar to [this](http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html)

Comment (riverszhang89) [inline bdb/file.c]: The instructions are correct. You must have mistyped something else.

**Comment by dorinhogea** [comment]
Instructions given:
09:51:31 RIVERS ZHANG : _ git diff master...<branch> -U0 --no-color | clang-format-diff -style="our style" -p1 -i

**Comment by dorinhogea** [comment]
followed; please update the tools used to check the repository in github

**Comment by riverszhang89** [comment]
Did you replace the placeholder `"our style"` with the actual clang format style?

**Comment by riverszhang89** [comment]
Everyone got it right so please stop complaining.:)

**Comment by dorinhogea** [comment]
tried to... maybe the checker should print out the full command to run? there would be no confusing or options mistyping

**Comment by riverszhang89** [comment]
Already has. See last roborivers comment.

**Comment by dorinhogea** [comment]
great, thank you, missed one link

**Comment by riverszhang89** [comment]
Anytime :smile:

**Pull request merged**